### PR TITLE
dragon block staff event

### PR DESCRIPTION
**Pull request by Zychronix**

### CODE REVIEW

File: src/main/java/legend/game/modding/events/battle/DragonBlockStaffEvent.java [hunk 1]
```diff
@@ -0,0 +1,6 @@
+package legend.game.modding.events.battle;
+
+import legend.game.modding.events.Event;
+
+public class DragonBlockStaffEvent extends Event {
```
Comment (LordMonoxide) [inline src/main/java/legend/game/modding/events/battle/DragonBlockStaffEvent.java:5]: Extend BattleEvent

File: src/main/java/legend/game/scripting/GameVarArrayParam.java [hunk 2]
```diff
@@ -615,7 +618,10 @@ private void writeCombatVar(final int index, final int val) {
       case 241 -> Scus94491BpeSegment_8006.battleState_8006e398._544 = val;
       case 242 -> Scus94491BpeSegment_8006.battleState_8006e398._548 = val;
       case 243 -> Scus94491BpeSegment_8006.battleState_8006e398._54c = val;
-      case 244 -> Scus94491BpeSegment_8006.battleState_8006e398.dragonBlockStaff_550 = val;
+      case 244 -> {
+        Scus94491BpeSegment_8006.battleState_8006e398.dragonBlockStaff_550 = val;
+        EVENTS.postEvent(new DragonBlockStaffEvent());
```
Comment (LordMonoxide) [inline src/main/java/legend/game/scripting/GameVarArrayParam.java:623]: I know retail doesn't do this but we need to check if it's being turned on or off and dispatch events accordingly

**Review by LordMonoxide — CHANGES_REQUESTED** [review]

File: src/main/java/legend/game/modding/events/battle/DragonBlockStaffEvent.java (hunk 1 above)
Comment (Zychronix) [reply src/main/java/legend/game/modding/events/battle/DragonBlockStaffEvent.java]: changes made

File: src/main/java/legend/game/scripting/GameVarArrayParam.java (hunk 2 above)
Comment (Zychronix) [reply src/main/java/legend/game/scripting/GameVarArrayParam.java]: I split the event, if the dragon block staff value is 1 it is on.

**Review by LordMonoxide — APPROVED** [review]

**Pull request merged**